app/V1Module/presenters/AssignmentSolutionsPresenter.php

```diff
@@ -189,15 +189,21 @@ public function checkSetBonusPoints(string $id) {
    * Set new amount of bonus points for a solution
    * @POST
    * @Param(type="post", name="bonusPoints", validation="numericint", description="New amount of bonus points, can be negative number")
+   * @Param(type="post", name="overriddenPoints", required=false, validation="numericint", description="Overrides points assigned to solution by the system")
    * @param string $id Identifier of the submission
+   * @throws NotFoundException
    */
   public function actionSetBonusPoints(string $id) {
     $newBonusPoints = $this->getRequest()->getPost("bonusPoints");
+    $overriddenPoints = $this->getRequest()->getPost("overriddenPoints");
     $solution = $this->assignmentSolutions->findOrThrow($id);
 
     $solution->setBonusPoints($newBonusPoints);
-    $this->assignmentSolutions->flush();
+    if ($overriddenPoints) {
```

Comment:
How do I remove the override (if the teacher changes hir mind)? I thought sending null should do that.

Reply:
... and 0 points is also a valid override.